src/main/ruby/core/range.rb

```diff
@@ -243,19 +243,17 @@ def min
 
   def step_internal(step_size=1) # :yields: object
     return to_enum(:step, step_size) do
-      m = Truffle::Mirror::Range.reflect(self)
-      m.step_iterations_size(*m.validate_step_size(self.begin, self.end, step_size))
+      Truffle::RangeOperations.step_iterations_size(self, *Truffle::RangeOperations.validate_step_size(self.begin, self.end, step_size))
```

Comment:
That's quite a long line, maybe extract the result of validate_step_size to a local variable?
Or maybe another helper in `Truffle::RangeOperations` can be added like `validate_step_iterations_size`.